sessionctx/stmtctx/stmtctx.go

```diff
@@ -70,6 +70,13 @@ type StatementContext struct {
 	InNullRejectCheck      bool
 	AllowInvalidDate       bool
 
+	// StartTime is the query start time.
+	StartTime time.Time
```

Comment:
All three need to be reset when retrying.

Reply:
1. `StartTime` and `DurationParse` should not be touched when retrying, because retry a txn doesn't parse the query.
2. `DurationCompile` should be summed up when retrying.

Correct me if I'm wrong.

Reply:
1. so it should be 0 when retrying?
1. it is a new transaction, I think it is better do not sum up them.

Reply:
> 1. so it should be 0 when retrying?
> 2. it is a new transaction, I think it is better do not sum up them.

I reset time duration for parse and compile to 0 when retrying. The query start time is also re-recorded when retrying.

I've changed the code logic to not sum them up. Another interesting behavior for retrying pessimistic txn and optimistic txn is, retrying a optimistic txn requires to rebuild the query execution plan, while retrying a pessimistic txn does not..

Reply:
I think pessimistic txn won't retry.